crates/aptos-ledger/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use hex::encode;
+use ledger_apdu::APDUCommand;
+use ledger_transport_hid::{hidapi::HidApi, TransportNativeHID};
+use std::str;
+
+const DERIVATIVE_PATH: &str = "m/44'/637'/0'/0'/0'"; // TODO: Add support for multiple index
+
+const CLA_APTOS: u8 = 0x5B; // Aptos CLA Instruction class
+const INS_GET_VERSION: u8 = 0x03; // Get version instruction code
+const INS_GET_APP_NAME: u8 = 0x04; // Get app name instruction code
+const INS_GET_PUB_KEY: u8 = 0x05; // Get public key instruction code
+const INS_SIGN_TXN: u8 = 0x06; // Sign the transaction
+const APDU_CODE_SUCCESS: u16 = 36864; // success code for transport.exchange
+
+const MAX_APDU_LEN: usize = 255;
+const P1_NON_CONFIRM: u8 = 0x00;
+const P1_CONFIRM: u8 = 0x01;
+const P1_START: u8 = 0x00;
+const P2_MORE: u8 = 0x80;
+const P2_LAST: u8 = 0x00;
+
+#[derive(Debug)]
+pub enum AptosLedgerError {
+    /// Error when trying to open a connection to the Ledger device
+    DeviceNotFound,
+    /// Unexpected error
+    UnexpectedError(String),
+}
```

Comment:
Let's use `thiserror` for this?

Reply:
Ah, so basically adding the `thiserror` crate, and define `#[error()]` for each error?